src/main/java/land/oras/credentials/FileStore.java

```diff
@@ -0,0 +1,166 @@
+package land.oras.credentials;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+import com.google.gson.reflect.TypeToken;
+import land.oras.exception.ConfigLoadingException;
+import land.oras.utils.JsonUtils;
+
+/**
+ * FileStore implements a credentials store using a configuration file
+ * to keep the credentials in plain-text.
+ *
+ * Reference: https://docs.docker.com/engine/reference/commandline/cli/#docker-cli-configuration-file-configjson-properties
+ */
+public class FileStore {
```

Comment:
Perhaps we should use package visibility and move it to `auth` package?

Not sure if we want to expose the FileStore directly on consumer. It's an internal logic

Reply:
I guess auth package should contains all the different types of auth process, Filestore it is kind of library to get and put credentials from the file store that's why i put in the different section

Reply:
The problem with plugin is that we will have consumer using it. So meaning we need to keep API compatible.

Reply:
okay got it @jonesbusy

Reply:
@jonesbusy If we use the ContainerRef, the file storage will looks like {ContainerRef: Credentials} and Credentials contains {username, password}, any thing extra we need to store in the file storage ?

Reply:
Not blocker for me, bind was worth to mentionnate